stake-pool/program/src/state.rs

```diff
@@ -534,6 +552,20 @@ pub struct Fee {
     pub numerator: u64,
 }
 
+impl Fee {
+    /// Applies the Fee's rates to a given amount, `amt`
+    /// returning the amount to be subtracted from it as fees
+    /// (0 if denominator is 0 or amt is 0),
+    /// or None if overflow occurs
+    pub fn apply(&self, amt: u128) -> Option<u128> {
```

Comment:
I like this!

nit: this should use `u64` as the interface. since the fee is <= 1, and amounts are only ever u64, it will fit.  If not, it'll correctly blow up